### PR TITLE
User profile edit + task difficulty + user difficulty setting

### PR DESCRIPTION
**Pull request by ad-world**

### CODE REVIEW

File: android/app/src/main/java/com/example/clarity/sdk/Utils.kt [hunk 1]
```diff
@@ -1,7 +1,8 @@
 package com.example.clarity.sdk
 
 data class SetMetadata(val set_id: Int, val title: String, val type: String, val is_public: Boolean, val likes: Int)
-data class UserWithId(val user_id: Int, val username: String, val email: String, val firstname: String, val lastname: String, val phone_number: String, val login_streak: Int)
+data class User(val username: String, val email: String, val password: String, val firstname: String, val lastname: String, val phone_number: String, val difficulty: Difficulty)
```
Comment (dhirpatel133) [inline android/app/src/main/java/com/example/clarity/sdk/Utils.kt:4]: is there a specific reason as to why we would use User object instead of UserWithId?

**Review by dhirpatel133 — APPROVED** [review]
everything else looks good. just that one comment I had above ^^

File: android/app/src/main/java/com/example/clarity/sdk/Utils.kt (hunk 1 above)
Comment (ad-world) [reply android/app/src/main/java/com/example/clarity/sdk/Utils.kt:4]: User object is used when creating a new user afaik, UserWithId is more of an object that represents a user in the db. Good question though, this is quite confusing so lemme j make sure where the User object is used and i'll make changes if required

Comment (ad-world) [reply android/app/src/main/java/com/example/clarity/sdk/Utils.kt]: fixed

**Pull request merged**